### PR TITLE
Implemented unsubscribe and handleUnsubscribe

### PR DESCRIPTION
**Pull request by ajw7**
Drafted implementation of unsubscribe (nonblocking) and handleUnsubscribe. No response is expected after sending an unsubscribe message.

### CODE REVIEW

File: core/src/edu/rice/rbox/Replication/ReplicaManagerGrpc.java [hunk 1]
```diff
@@ -151,7 +174,20 @@ public void handleSubscribe(Rbox.SubscribeRequest request,
 
         @Override
         public void handleUnsubscribe(Rbox.UnsubscribeRequest request, StreamObserver<Empty> responseObserver) {
-            // TODO: Remove from subscribers
+            logger.info("Handling unsubscribe request...");
+
+            GameObjectUUID replicaObjectUUID = getGameObjectUUIDFromMessage(request.getMsg());
+            GameObjectUUID primaryObjectUUID = publishers.get(replicaObjectUUID).getGameObjectUUID();
```
Comment (isabella-h) [inline core/src/edu/rice/rbox/Replication/ReplicaManagerGrpc.java]: The `replicaObjectUUID` in your `UnsubscribeRequest` is the uuid of the replica object on the sender superpeer, while `publishers` is mapping replicas on this receiver superpeer to their primaries. So you cannot get the corresponding primary object uuid from `publishers.get(replicaObjectUUID)`.

**Comment by vhellmer** [comment]
looks good to me

**Comment by vhellmer** [comment]
wait, sorry - I accidentally closed this, my bad

File: core/src/edu/rice/rbox/Replication/ReplicaManagerGrpc.java (hunk 1 above)
Comment (isabella-h) [reply core/src/edu/rice/rbox/Replication/ReplicaManagerGrpc.java]: Wait, I was assuming replica and primary objects have different `GameObjectUUID`. Is that correct?

**Comment by ajw7** [comment]
Updated since clarifying that primary and replica object UUIDs are the same.

**Pull request merged**